src/core/room-lifecycle-manager.ts

```diff
@@ -1,923 +1,410 @@
-      .catch((error: unknown) => {
-        this._logger.error('RoomLifecycleManager._onChannelSuspension(); unexpected error thrown', { error });
-      });
   }
 
   /**
-   * Given some contributor that has entered a suspended state:
-   *
-   * - Wind down any other channels
-   * - Wait for our contributor to recover
-   * - Attach everything else
-   *
-   * Repeat until either of the following happens:
-   *
-   * - Our contributor reattaches and we can attach everything else (repeat with the next contributor to break if necessary)
-   * - The room enters a failed state
-   *
-   * @param contributor The contributor that has entered a suspended state.
-   * @returns A promise that resolves when the room is attached, or the room enters a failed state.
+   * Sets up monitoring for channel discontinuities.
+   * A discontinuity exists when an attached or update event comes from the channel with resume=false.
+   * The first time we attach, or if we attach after an explicit detach call are not considered discontinuities.
+   * @private
    */
-  private async _doRetry(contributor: ContributesToRoomLifecycle): Promise<void> {
-    // A helper that allows us to retry the attach operation
-    // eslint-disable-next-line unicorn/consistent-function-scoping
-    const doAttachWithRetry = () => {
-      this._logger.debug('RoomLifecycleManager.doAttachWithRetry();');
-      this._lifecycle.setStatus({ status: RoomStatus.Attaching });
-
-      return this._doAttach().then((result: RoomAttachmentResult) => {
-        this._logger.debug('RoomLifecycleManager.doAttachWithRetry(); attach result', {
-          status: result.status,
-          error: result.error,
-          failedFeature: result.failedFeature?.channel.name,
-        });
-        // If we're in failed, then we should wind down all the channels, eventually - but we're done here
-        if (result.status === RoomStatus.Failed) {
-          void this._mtx.runExclusive(
-            () =>
-              this._runDownChannelsOnFailedAttach().finally(() => {
-                this._endLifecycleOperation();
-              }),
-            LifecycleOperationPrecedence.Internal,
-          );
-          return;
-        }
-
-        // If we're in suspended, then we should wait for the channel to reattach and then try again
-        if (result.status === RoomStatus.Suspended) {
-          const failedFeature = result.failedFeature;
-          if (!failedFeature) {
-            throw new Ably.ErrorInfo('no failed feature in _doRetry', ErrorCodes.RoomLifecycleError, 500);
-          }
-
-          this._logger.debug('RoomLifecycleManager.doAttachWithRetry(); feature suspended, retrying attach', {
-            feature: failedFeature.channel.name,
-          });
-          return this._doRetry(failedFeature).catch();
-        }
-
-        // We attached, huzzah! It's the end of the loop
-        this._endLifecycleOperation();
-      });
-    };
+  private _startMonitoringDiscontinuity(): void {
```

Comment:
We don't need this method as per https://github.com/ably/ably-chat-js/pull/505/files#r2034471154.
This method has duplicated code and two additional listeners.